### DISCUSSION TITLE
Valid characters for a trino table

### DISCUSSION THREAD

**Discussion by curious-bug** [trinodb/trino#20944] [Q&A]
Trino Version: 380
Connector: Hive

Is there a definitive list of what characters are and are not valid for table/view names in trino?
I am writing a simple check function in python to validate that a table name is safe to use (free of SQL injection, etc) and haven't found anything online.

**Comment by mosabua** [comment]
I am not sure there is a definitive list but I would stick with lowercase alphanumerics and maybe _ .. and thats it. More might work but is bound to cause headaches in different parts of the stack

**Reply by curious-bug** [reply]
That's what I've done so far. I created the regular expression
```regex
^(?:[a-z_]+\.){0,2}([a-z_]+|(?:"[a-z_]+\$[a-z_]+"))$
```
That matches to a valid table name including if the schema and catalog are specified and if its a hive metadata table.
I'm just curious if theres more characters I should add as valid.

**Reply by mosabua** [reply]
There might be .. but why are you trying to find more? Is that not enough? Which ones work might end up varying a lot on.  What is the actual use case for that function so that it is not enough what you have now?

**Reply by curious-bug** [reply]
It is enough for my use case. I just curious if there should be more for a "proper"  check

**Reply by hashhar** [reply]
In case of Hive the metastore is also a variable. Trino might accept something but the metastore migth not. It can also vary from version to version.

**Reply by hashhar** [reply]
What actually are you trying to do? Why not just try to do the operation and then see if it works or fails?

Or as Manfred recommended, stick with simple set of alphanumerics + `_`. Special characters will need to be quoted anyway like `"my-table"` so it's additional work.

**Comment by mosabua** [comment]
The other thing worth mentioning is that case sensitivity is also inconsistent across data sources and not consistently supported by Trino either.. hence the recommendation to stick to lowercase...
